persistence/src/main/java/de/cyface/persistence/Persistence.java

```diff
@@ -0,0 +1,682 @@
+package de.cyface.persistence;
+
+import static de.cyface.persistence.Constants.TAG;
+import static de.cyface.persistence.serialization.MeasurementSerializer.TRANSFER_FILE_FORMAT_VERSION;
+import static de.cyface.persistence.serialization.MeasurementSerializer.serialize;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.zip.Deflater;
+
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.util.Log;
+import androidx.annotation.NonNull;
+import de.cyface.persistence.model.GeoLocation;
+import de.cyface.persistence.model.Measurement;
+import de.cyface.persistence.model.Vehicle;
+import de.cyface.persistence.serialization.*;
+import de.cyface.utils.Validate;
+
+/**
+ * This class wraps the Cyface Android persistence API as required by the <code>DataCapturingService</code>,
+ * <code>SyncAdapter</code> and its
+ * delegate objects.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.2
+ * @since 3.0.0
+ */
+public class Persistence {
+
+    /**
+     * <code>ContentResolver</code> that provides access to the {@link MeasuringPointsContentProvider}.
+     */
+    private final ContentResolver resolver;
+    /**
+     * The authority used to identify the Android content provider to persist data to or load it from.
+     */
+    private final String authority;
+    /**
+     * The {@link File} pointing to the directory containing open measurements.
+     */
+    protected final File openMeasurementsDir;
```

Comment:
Don't you have wrapper classes like MeasurementFile, which you should use here?

Reply:
it's now in the Measurement class and FileUtils (to generate the names)